easybuild/framework/easyconfig/tweak.py

```diff
@@ -833,6 +950,71 @@ def map_easyconfig_to_target_tc_hierarchy(ec_spec, toolchain_mapping, targetdir=
                 # set module names
                 orig_dep['short_mod_name'] = ActiveMNS().det_short_module_name(dep)
                 orig_dep['full_mod_name'] = ActiveMNS().det_full_module_name(dep)
+            elif update_dep_versions:
+                # Search for available updates for this dependency:
+                # First get all candidate paths for this (include search through subtoolchains)
+                toolchain_hierarchy = get_toolchain_hierarchy(orig_dep['toolchain'])
+                potential_version_matches = []
+
+                # Figure out what precedes the version
+                if 'versionprefix' in dep:
+                    versionprefix = dep['versionprefix']
+                else:
+                    versionprefix = ''
+                versionprefix = ''.join([x for x in [dep['name'] + '-', versionprefix] if x])
+                # Figure out the main versionsuffix (altered depending on toolchain in the loop below)
+                if 'versionsuffix' in dep:
+                    if dep['versionsuffix'] in versonsuffix_mapping:
+                        versionsuffix = versonsuffix_mapping[dep['versionsuffix']]
+                    else:
+                        versionsuffix = dep['versionsuffix']
+                else:
+                    versionsuffix = ''
+
+                for toolchain in toolchain_hierarchy:
+                    candidate_ver = '.*'  # using regex for *
+
+                    # determine main install version based on toolchain
+                    if toolchain['name'] != DUMMY_TOOLCHAIN_NAME:
+                        toolchain_suffix = "-%s-%s" % (toolchain['name'], toolchain['version'])
+                    else:
+                        toolchain_suffix = ''
+                    full_versionsuffix = ''.join([x for x in [toolchain_suffix, versionsuffix, EB_FORMAT_EXTENSION]
```

Comment:
Hmm, is `EB_FORMAT_EXTENSION` really needed? If so, it should be handled somewhere else, I think...

Reply:
Well, I wanted to know that there's nothing else coming after when I do the search below, I can move it down there